src/launchpad/insights/insight.py

```diff
@@ -0,0 +1,35 @@
+from dataclasses import dataclass
+from typing import Protocol, TypeVar
+
+from ..models.apple import AppleAppInfo, MachOBinaryAnalysis
+from ..models.common import FileAnalysis
+from ..models.treemap import TreemapResults
+
+T_co = TypeVar("T_co", covariant=True)
+
+
+@dataclass
+class InsightsInput:
+    app_info: AppleAppInfo
+    file_analysis: FileAnalysis
+    treemap: TreemapResults | None
+    binary_analysis: list[MachOBinaryAnalysis]
+
+
+class Insight(Protocol[T_co]):
+    """Protocol for insight functions.
+
+    Insights are functions that take analysis results and return typed insight results.
+    All data needed for the insight must be collected during the main analysis phase.
+    """
+
+    def get_results(self, input: InsightsInput) -> T_co:
```

Comment:
nit: could this be something like `perform()` to indicate that work is being done, `get` kind of hides that.

Reply:
e.g. when I implement the image compression insight that becomes more apparent